src/main/java/org/opensearch/flowframework/workflow/RegisterModelGroupStep.java

```diff
@@ -142,7 +143,7 @@ public void onFailure(Exception ex) {
             String modelGroupName = (String) inputs.get(NAME_FIELD);
             String description = (String) inputs.get(DESCRIPTION_FIELD);
             List<String> backendRoles = getBackendRoles(inputs);
-            AccessMode modelAccessMode = (AccessMode) inputs.get(MODEL_ACCESS_MODE);
+            AccessMode modelAccessMode = AccessMode.from(((String) inputs.get(MODEL_ACCESS_MODE)).toLowerCase(Locale.ROOT));
```

Comment:
Since `MODEL_ACCESS_MODE` is optional, there's a chance the `get()` result is `null` here.  Casting a null Object to a String, it's still null.

AccessMode.from(null) appears to return null, which would throw an NPE on trying to lowercase it.

Reply:
Wouldn't ml-commons would treat `AccessMode` as public when passed null? @amitgalitz

Reply:
its public if security isn't enabled and private when not enabled so we should pass null when its null and when given a value we should pass correct lower case string